tests/test_history.py

```diff
@@ -383,62 +458,64 @@ def hist_file():
     except FileNotFoundError:
         pass
 
-def test_existing_history_file(hist_file, capsys):
-    import atexit
-    import readline
-
-    # Create the history file before making cmd2 app
-    with open(hist_file, 'w'):
-        pass
+def test_bad_history_file_path(capsys, request):
+    # Use a directory path as the history file
+    test_dir = os.path.dirname(request.module.__file__)
```

Comment:
This won't work on all of our CI systems due to permission issues - specifically it is failing on AppVeyor.  I recommend changing it to use a tempdir/tempfile since that should be writeable on all of our CI systems.

Reply:
Fixed. Now using `tempfile.TemporaryDirectory()` as a context manager. around the test code.